src/qtfirebaseadmob.cpp

```diff
@@ -888,6 +888,432 @@ void QtFirebaseAdMobBanner::moveTo(int position)
+        if(future.Error() != admob::kAdMobErrorNone) {
+            qDebug() << this << "::setY ERROR code" << future.Error() << "message" << future.ErrorMessage();
+            return;
+        }
+        */
+        qDebug() << this << "::setY moved to" << _x << "," << y;
+        _y = y;
+        emit yChanged();
+    }
+}
+
+int QtFirebaseAdMobNativeAd::getWidth()
+{
+    return _width;
+}
+
+void QtFirebaseAdMobNativeAd::setWidth(const int &width)
+{
+    if(_width != width) {
+        _width = width;
+        emit widthChanged();
+    }
+}
+
+int QtFirebaseAdMobNativeAd::getHeight()
+{
+    return _height;
+}
+
+void QtFirebaseAdMobNativeAd::setHeight(const int &height)
+{
+    if(_height != height) {
+        _height = height;
+        emit heightChanged();
+    }
+}
+
+QtFirebaseAdMobRequest* QtFirebaseAdMobNativeAd::request() const
+{
+    return _request;
+}
+
+void QtFirebaseAdMobNativeAd::setRequest(QtFirebaseAdMobRequest* request)
+{
+    if(_request != request) {
+        _request = request;
+        emit requestChanged();
+    }
+}
+
+void QtFirebaseAdMobNativeAd::init()
+{
+    //qDebug() << "QtFirebase.AdMobBanner pre-initialize _ready" << _ready << "_init" << _initializing;
+    if(!qFirebase->ready()) {
+        qDebug() << this << "::init" << "base not ready";
+        return;
+    }
+
+    if(!qFirebaseAdMob->ready()) {
+        qDebug() << this << "::init" << "AdMob base not ready";
+        return;
+    }
+
+    if(_adUnitId.isEmpty()) {
+        qDebug() << this << "::init" << "adUnitId must be set in order to initialize the banner";
+        return;
+    }
+
+    if(_isFirstInit && !PlatformUtils::getNativeWindow()) {
+        qDebug() << this << "::init" << "no native ui element. Waiting for it...";
+        return;
+    }
+
+    // TODO test nessecity of this
+    if(!_nativeUIElement && !PlatformUtils::getNativeWindow()) {
+        qDebug() << this << "::init" << "no native ui element";
+        return;
+    }
+
+    if(!_nativeUIElement && PlatformUtils::getNativeWindow()) {
+        qDebug() << this << "::init" << "setting native ui element";
+        _nativeUIElement = PlatformUtils::getNativeWindow();
+    }
+
+    if(!_ready && !_initializing) {
+        _initializing = true;
+
+        _nativeAd = new admob::NativeExpressAdView();
+
+        admob::AdSize ad_size;
+        ad_size.ad_size_type = admob::kAdSizeStandard;
+        ad_size.width = getWidth();
+        ad_size.height = getHeight();
+
+        // A reference to an iOS UIView or an Android Activity.
+        // This is the parent UIView or Activity of the banner view.
+        qDebug() << this << "::init initializing with AdUnitID" << __adUnitIdByteArray.constData();
+        firebase::FutureBase future = _nativeAd->Initialize(static_cast<admob::AdParent>(_nativeUIElement), __adUnitIdByteArray.constData(), ad_size);
+        qDebug() << this << "::init" << "native initialized";
+        qFirebase->addFuture(__QTFIREBASE_ID + ".nativead.init",future);
```

Comment:
Let's keep it a bit more consistent :)
`qFirebase->addFuture(__QTFIREBASE_ID + ".nativeexpressad.init",future);`